### PR TITLE
use same version vstsom with vstshost.

### PR DESCRIPTION
**Pull request by TingluoHuang**

### CODE REVIEW

**Comment by TingluoHuang** [comment]
```
Task 'CopyPublishBuildArtifacts' (1.0.32) is using deprecated task execution handler, stop use this task and contact task author to upgrade their task.
```

File: src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs [hunk 1]
```diff
@@ -177,6 +177,9 @@ public async Task RunAsync()
             ArgUtil.NotNull(Inputs, nameof(Inputs));
             ArgUtil.Directory(TaskDirectory, nameof(TaskDirectory));
 
+            // Warn about legacy handler.
+            ExecutionContext.Warning($"Task '{this.Task.Name}' ({this.Task.Version}) is using deprecated task execution handler, stop use this task and contact task author to upgrade their task.");
```
Comment (bryanmacfarlane) [inline src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs]: Task '{this.Task.Name}' ({this.Task.Version}) is using deprecated task execution handler.  The task should use the supported task-lib: https://aka.ms/tasklib

Comment (bryanmacfarlane) [reply src/Agent.Worker/Handlers/LegacyPowerShellHandler.cs]: Date per Matt Cooper

File: src/Misc/layoutroot/config.cmd [hunk 2]
```diff
@@ -23,7 +23,7 @@ rem   Operation is not supported.
...
...-Command "$VerbosePreference = %VERBOSE_ARG% ; Get-ChildItem -LiteralPath '%~dp0' | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null ; Get-ChildItem -Recurse -LiteralPath '%~dp0bin', '%~dp0externals' | Where-Object { $_ -match '\.(ps1|psd1|psm1)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null ; if (Test-Path -LiteralPath '%~dp0externals\vstsom' -PathType Container) { Get-ChildItem -LiteralPath '%~dp0externals\vstsom' | Where-Object { $_ -match '\.(dll|exe)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null } if (Test-Path -LiteralPath '%~dp0externals\vstshost' -PathType Container) { Get-ChildItem -LiteralPath '%~dp0externals\vstshost' | Where-Object { $_ -match '\.(dll|exe)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null } if (Test-Path -LiteralPath '%~dp0externals\tf' -PathType Container) { Get-ChildItem -LiteralPath '%~dp0externals\tf' | Where-Object { $_ -match '\.(dll|exe)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null }"
```
Comment (ericsciple) [inline src/Misc/layoutroot/config.cmd]: whats the difference?

File: src/Agent.Plugins/TFCliManager.cs [hunk 3]
```diff
@@ -33,11 +33,11 @@ public override TfsVCFeatures Features
 
         protected override string Switch => "/";
 
-        public string FilePath => Path.Combine(ExecutionContext.Variables.GetValueOrDefault("Agent.ServerOMDirectory")?.Value, "tf.exe");
+        public string FilePath => Path.Combine(ExecutionContext.Variables.GetValueOrDefault("Agent.HomeDirectory")?.Value, "externals", "tf", "tf.exe");
```
Comment (ericsciple) [inline src/Agent.Plugins/TFCliManager.cs]: - do we ever add tf to the path?
- do we expect customers are looking for tf in a well known location? will that break or is it still in the om directory too?

File: src/Misc/layoutroot/config.cmd (hunk 2 above)
Comment (TingluoHuang) [reply src/Misc/layoutroot/config.cmd]: unblock file under `dp0externals\tf`

**Pull request merged**